slang-gfx.h

```diff
@@ -544,6 +544,13 @@ class IShaderObject : public ISlangUnknown
         setSampler(ShaderOffset const& offset, ISamplerState* sampler) = 0;
     virtual SLANG_NO_THROW Result SLANG_MCALL setCombinedTextureSampler(
         ShaderOffset const& offset, IResourceView* textureView, ISamplerState* sampler) = 0;
+
+        /// Manually setting the specialization arguments for the shader object, overriding
+        /// the default arguments computed from the sub-objects.
+        /// Specialization arguments are passed to the shader compiler to specialize the type
+        /// of interface-typed shader parameters.
+    virtual SLANG_NO_THROW Result SLANG_MCALL
+        setSpecializationArgs(const slang::SpecializationArg* args, uint32_t count) = 0;
```

Comment:
Unrelated to this specific PR, but: it seems like we probably need/want to have a distinct `SpecializationArg::Kind` to represent what we currently use `__Dynamic` for. That would make it so that a user could request dynamic dispatch *without* needing to look up the `__Dynamic` type.

Reply:
If you want to just specialize for a sub-object range, you can still just call `setSpecializationArgs` on that sub-object only.

Reply:
Hmm... I'm not 100% convinced that is correct. Even so, setting specialization state on a sub-object itself creates the problem that now the *object* is what is deciding whether or not to specialize: it is either always specialized or never specialized. In that model you can't bind the same sub-object to shaders where you want specialization and shaders where you want dynamic dispatch.

Maybe I'm not understanding why encoding specialization arguments as a whole-object thing instead of a per-subobject-range thing is a Good Thing. I can see a strong argument for per-range control, but I don't see anything yet that argues for why the whole-object case should be preferred.